internal/oracle/oracle.go

```diff
@@ -52,6 +52,8 @@ func (o *Oracle) FindDefinition(q DefinitionQuery) (*DefinitionQueryResult, erro
 	// TODO(tsandall): how can we cache the results of compilation and parsing so that
 	// multiple queries can be executed without having to re-compute the same values?
 	// Ditto for caching across runs. Avoid repeating the same work.
+
+	// NOTE(sr): "SetRuleTree" because it's needed for compiler.GetRulesExact() below
```

Comment:
So the early writing steps had broken the oracle -- it depended on the fact that no generated expression has made it into the rule bodies yet to determine the span of a body (L222). That calculation has been adapted below, to ignore generated expressions trailing the body.